File: November 2022/13 November 2022/Java/Fill up buckets.java
//{ Driver Code Starts
//Initial Template for Java

import java.util.*;
import java.lang.*;
import java.io.*;
class GFG
{
    public static void main(String[] args) throws IOException
    {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine().trim());
        while(T-->0)
        {
            int n = Integer.parseInt(br.readLine().trim());
            String s = br.readLine().trim();
            String[] S = s.split(" ");
            int[] capacity = new int[n];
            for(int i = 0; i < n; i++)
                capacity[i] = Integer.parseInt(S[i]);
            Solution ob = new Solution();
            int ans = ob.totalWays(n, capacity);
            System.out.println(ans);
        }
    }
}

// } Driver Code Ends


//User function Template for Java

class Solution{
    int mod=(int) (1e9 + 7);
    public int totalWays(int n, int[] capacity) {
        // code here
        if(n==1) return capacity[0];
        Arrays.sort(capacity);
        long res=1;
        for(int i=0;i<n;i++)
            res=(res%mod)*(capacity[i]-i)%mod;
        return (int)res;
    }
}

File: November 2022/13 November 2022/Python/Fill up buckets.py
#User function Template for python3

class Solution:
	def totalWays(self, n, capacity):
        capacity.sort()
        ans, MOD = 1, 1000000007
        
        for i, cap in enumerate(capacity):
            ans = (ans * (cap - i)) % MOD
        return ans

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
	T=int(input())
	for i in range(T):
		n= int(input())
		capacity = list(map(int,input().split()))
		ob = Solution()
		ans = ob.totalWays(n,capacity)
		print(ans)

# } Driver Code Ends